pkg/operator/ceph/cluster/cluster.go

```diff
@@ -67,6 +69,37 @@ type childController interface {
 	ParentClusterChanged(cluster cephv1.ClusterSpec, clusterInfo *cephconfig.ClusterInfo)
 }
 
+// handlerCtx is the context used by the add and update handlers of a cluster.
+// This can be used to cancel any running handlers for a cluster
+type handlerCtx struct {
+	context.Context
+	cancel  context.CancelFunc
+	wait    chan struct{}
+	success bool
+}
+
+// newHandlerCtx returns a handlerCtx with the given timeout
+func newHandlerCtx(timeout time.Duration) *handlerCtx {
+	var (
```

Comment:
nit: could you change this to just be 
```go
var ctx context.Context
var cancel context.CancelFunc
```

Including only two vars in a var block like this adds the appearance of additional cyclomatic complexity to the function as well as adding lines.